### ISSUE TITLE
Display DOI link in bibliography

### ISSUE THREAD

**Issue by hellman** [Cryptosaurus/iacrtrans#37]
Currently, the class does not display DOIs in bibliography, even if they are given in bib files (like in the cryptobib database). I think DOIs are important for indexation purposes (and also convenient to click directly). A possible downside is that more space is taken and this would affect venues which include bibliography in page count. 

Would be nice to have an option for displaying clickable DOIs in biblio.

**Comment by kmccurley** [comment]
We use the `alphaurl` bibliography style in iacrcc in order to make them clickable. This is part of the standard texlive distribution and is just a variation of the alpha style.

I'm in the process of writing a new bibtex => HTML translator, so I've been studying bibtex a lot. One reason for doing this is to check the bibtex that is submitted for the CiC journal, and it's really difficult to validate that the bibtex references are properly formatted. There can be several problems:
1. some people use the https://doi.org/ in their DOIs, but the DOI is supposed to start with 10. and the https part doesn't belong there. This can be checked with the l3regex package of latex.
2. some people include url = {https://doi.org/...} instead of the DOI field. The url field should be use for a non-DOI url.
3. some people use note = {\url{https://doi.org/...}}
4. some people use howpublished = {\url{https://doi.org/...}}
5. many people have been trained to strip DOIs from their bibtex because they are trying to fit their paper into 30 pages and the references used to be counted. The IACR board adopted a resolution that references should no longer count in the page limit. In a purely online journal like ToSC and TCHES this should not arise but it's hard to change author habits.

**Comment by hellman** [comment]
Oh, I didn't know about `alphaurl`, thanks! This is exactly what I was looking for.

Should it be the default then for iacrtrans? E.g. in [./template.tex](../blob/master/template.tex#L76) ? (I can create a PR if makes sense)

I haven't seen papers using it in TCHES/ToSC (even the final versions).

**Comment by Cryptosaurus** [comment]
Yes, I agree it would be good to recommend a style that includes the DOI. This should be mentioned in the documentation, and used in the template. The `alphaurl` style seems to be a good candidate (another option is to switch to `biblatex` which is more modern and can be configured by the LaTeX file, but that's a more radical change).

More generally, it would be good to do more work on how to do references for ToSC/TCHES/CiC, but I never took the time to do it seriously. The current recommendation for `iacrtrans` is to use bibliography files from https://cryptobib.di.ens.fr/ or from https://dblp.org/. These cover most of the references required in Crypto papers, and they are generally of good quality (Cryptobib typically fixes a few mistakes from DBLP).

DBLP entries include a DOI, both the `doi` field and in the `url` field; the `alphabst` style (*Edit*: I mean `alphaurl`) shows them correctly. Most cryptobib entries have a `doi` field, but some papers are missing it (eg: "The Wide Trail Design Strategy" [IMA:DaeRij01]). This seems to be a known issue with Cryptobib (https://github.com/cryptobib/db/issues/32), which could be fixed by writing a script to extract DOIs from DBLP and add them to the Cryptobib database.

**Comment by hellman** [comment]
By `alphabst` do you mean the standard `alpha.bst` (`\bibliographystyle{alpha}`)? As far as I can see it does not render the `doi` field. While `alphaurl` does (as hyperlink `doi:10.100...`).

**Comment by Cryptosaurus** [comment]
I mean `alphaurl`, sorry for the stupid mistake.

**Comment by kmccurley** [comment]
There is also the `plainurl` style - it depends on what you prefer for the standard look of a bibliography.

The last time I looked, there were over a thousand entries in cryptobib that were missing a DOI. I had a side project to back fill them, and I was using the crossref API to look up an article by title and author names. Unfortunately this is error-prone and I never finished the project. Now that CiC is almost launched, I may return to finish this.

Another good thing to recommend is the `xurl` package that will clealy break URLs across lines. URLs are often long and the LaTeX hyphenation algorithm has trouble with them.

biblatex has more features and is more modern, but it's [not without flaws](https://github.com/plk/biblatex/issues). It turns out that `biber` and `bibtex` have different behavior when they encounter keys that differ only in case (both entry keys and field keys). They also have different notions of what are required keys for an entry type.

**Comment by Cryptosaurus** [comment]
Since `iacrtrans` currently uses the `alpha` style, it makes more sense to move to `alphaurl` than to `plainurl`.

Regrading cryptobib, my understanding if they use DBLP as a source, but some conferences did not have a DOI in DBLP when they imported them, and have them now. I guess it would be more reliable to back fill them from the current DBLP than from crossref. (I actually started writing code for that many years ago...)

For breaking URLs I sometimes add `\usepackage[hyphens]{url}` in my papers, and it is usually sufficient. It seems that xurl does something similar but with even more allowed more break points. I don't now what provides the best results...

Regarding biblatex, I like to use it in some of my documents, but I agree with your comments :-)

**Comment by kmccurley** [comment]
I recently added a large number of DOIs to cryptobib, so it now has good coverage. There are some things that lack DOIs, including NDSS and Usenix proceedings, but that's because the publisher does not assign them. Note that people can now [search cryptobib](https://publish.iacr.org/cryptobib) and retrieve individual fully-expanded entries without @string or @crossref.

We have no plans to add DOIs to eprint, because we would prefer that people cite the official peer-reviewed versions unless the eprint version contains something not in the peer-reviewed version.

Now that cic.iacr.org has published two issues, you will notice that we extract the references from the paper so that we can do several things:
1. identify problems for copy editing.
2. show the references in HTML on the landing page for a paper.
3. most importantly we now report references to crossref.org when we register the DOI for a paper. I've noticed that if a reference is lacking a DOI, then crossref has a hard time matching it.
